mllib/src/main/scala/org/apache/spark/ml/Pipeline.scala

```diff
@@ -297,7 +297,9 @@ object Pipeline extends MLReadable[Pipeline] {
       val stageUids: Array[String] = (metadata.params \ "stageUids").extract[Seq[String]].toArray
       val stages: Array[PipelineStage] = stageUids.zipWithIndex.map { case (stageUid, idx) =>
         val stagePath = SharedReadWrite.getStagePath(stageUid, idx, stageUids.length, stagesDir)
-        val reader = DefaultParamsReader.loadParamsInstanceReader[PipelineStage](stagePath, spark)
```

Comment:
We had the same code in branch-4.0.

https://github.com/apache/spark/blob/ba029a88f6bb13ca8a52c4b8343951fb2e449dd1/mllib/src/main/scala/org/apache/spark/ml/Pipeline.scala#L300

Do we need to fix in `branch-4.0` too, @WeichenXu123 ?

<img width="569" alt="Screenshot 2025-05-13 at 20 58 13" src="https://github.com/user-attachments/assets/5107e2f1-04a8-4940-a757-be5dd3000da5" />

Reply:
FYI, cc @cloud-fan as the release manager of Apache Spark 4.0.0 .